src/main/java/org/kritiniyoga/karmayoga/allocators/JobSequencingAllocator.java

```diff
@@ -0,0 +1,44 @@
+package org.kritiniyoga.karmayoga.allocators;
+
+import io.vavr.collection.List;
+import io.vavr.collection.Seq;
+import org.kritiniyoga.karmayoga.Allocator;
+import org.kritiniyoga.karmayoga.Schedule;
+import org.kritiniyoga.karmayoga.Task;
+import org.kritiniyoga.karmayoga.TimeSlot;
+
+import java.util.Comparator;
+
+public class JobSequencingAllocator implements Allocator {
+
+    @Override
+    public Seq<Schedule> allocate(Seq<Task> tasks, Seq<TimeSlot> slots) {
+
+        List<Schedule> schedules = List.of();
+        for (Task task : sort(tasks)) {
+            TimeSlot slot = getSlot(slots, task);
+            slots = slots.remove(slot);
+            if (slot != null) {
+                Schedule schedule = Schedule.createSchedule(slot, task);
+                schedules = schedules.append(schedule);
+            }
+
+        }
+        return schedules;
+
+    }
+
+    private TimeSlot getSlot(Seq<TimeSlot> slots, Task task) {
```

Comment:
rename method? The method is obviously doing a bit of logic which isn't reflected in the name.